packages/sidenav/src/sidenav.ts

```diff
@@ -36,6 +39,88 @@ export class SideNav extends LitElement {
         this.value = event.detail.value;
     }
 
+    public constructor() {
+        super();
+        this.addEventListener('focusin', this.handleFocusin);
+        this.addEventListener('focusout', this.stopListeningToKeyboard);
+    }
+
+    private handleFocusin = (event: FocusEvent): void => {
+        this.startListeningToKeyboard();
+        if (event.target !== this) {
+            return;
+        }
+        const selected = this.querySelector('[selected]') as SideNavItem;
+        if (selected && !this.isDisabledChild(selected)) {
+            selected.focus();
+            return;
+        }
+        const items = [...this.querySelectorAll('sp-sidenav-item')];
+        let index = 0;
+        while (
+            index < items.length &&
+            items[index] &&
+            this.isDisabledChild(items[index])
+        ) {
+            index += 1;
+        }
+        /* istanbul ignore else */
+        if (items[index]) {
+            items[index].focus();
+        }
+    };
+
+    private startListeningToKeyboard = (): void => {
+        this.addEventListener('keydown', this.handleKeydown);
+    };
+
+    private stopListeningToKeyboard = (): void => {
+        this.removeEventListener('keydown', this.handleKeydown);
+    };
+
+    private handleKeydown(event: KeyboardEvent): void {
+        const { code } = event;
+        /* istanbul ignore if */
+        if (code !== 'ArrowDown' && code !== 'ArrowUp') {
+            return;
+        }
+        event.preventDefault();
+        const direction = code === 'ArrowDown' ? 1 : -1;
+        this.focusItemByOffset(direction);
+    }
+
+    private focusItemByOffset(direction: number): void {
+        const items = [...this.querySelectorAll('sp-sidenav-item')];
+        const focused = items.indexOf(document.activeElement as SideNavItem);
+        let next = focused;
+        next = (items.length + next + direction) % items.length;
+        while (this.isDisabledChild(items[next])) {
+            next = (items.length + next + direction) % items.length;
+        }
+        items[next].focus();
+    }
+
+    private isDisabledChild(child: SideNavItem): boolean {
```

Comment:
A child is disabled if `this.disabled`, a parent is `parent.disabled`, or a parent is `!parent.expanded`.

Reply:
Why do we search up through the dom to determine disabled state instead of propagate disabled state down?

Reply:
This maybe less clear from the outside because `disabled` isn't the only thing that's being checked here. It's also checking is a parent is `expanded`, that way maybe this should be `isAccessibleChild` or `isFocusableChild`?  Because `parent` can be a number of different element types and they all exist in a light DOM series none of them manages their children at current. We could acquire the slotted elements, though we might actually need a deep mutation observer for the reality that there can be intermediary `sp-sidenav-heading` elements, and work with altering the state of items imperatively, but that seemed like a lot of prep work that might not be needed, whereas this process this reality lazily. Along with that, if parents contextually managed the `disabled` state of an item, then we'd also have to manage a cache of the "expected" state to return to, which seemed like a lot of extraneous bookkeeping. Let me know if you have other thoughts that could shift the weight of one side of the balance.